ethcore/light/src/types/request/mod.rs

```diff
@@ -90,6 +90,8 @@ pub enum ResponseError<T> {
 	Validity(T),
 	/// No responses expected.
 	Unexpected,
+	/// Empty response
+	EmptyResponse,
```

Comment:
How is it different from `ResponseError::Validity(ValidityError::Empty)`?

Reply:
Excellent question, this is a bit blurry

- ```ResponseError:EmptyResponse``` - A response that is found `empty` before ```on_supply_responses``` https://github.com/paritytech/parity-ethereum/blob/master/ethcore/light/src/on_demand/mod.rs#L610-#L618
- ```ResponseError::Validity(ValidityError::Empty)``` - A response that validated by ```check_response``` https://github.com/paritytech/parity-ethereum/blob/master/ethcore/light/src/on_demand/request.rs#L620

Shall I change only to use ```ResponseError::Validity(ValidityError::Empty)```?